pkg/elb/cleanup.go

```diff
@@ -130,6 +171,38 @@ func getServiceLoadBalancer(ctx context.Context, ec2API ec2iface.EC2API, elbAPI
 	return &lb, nil
 }
 
+func getIngressLoadBalancer(ctx context.Context, ingress networkingv1beta1.Ingress) (lb *loadBalancer) {
+	ingressCls := "kubernetes.io/ingress.class"
+	if ingress.Annotations[ingressCls] != "alb" {
+		logger.Debug("%s is not ALB Ingress, it is '%s': '%s', skip", ingress.Name, ingressCls, ingress.Annotations[ingressCls])
+		return nil
+	}
+
+	// Check if field status.loadBalancer.ingress[].hostname is set, value corresponds with name for AWS ALB
+	// if does not pass ALB hadn't been provisioned so nothing to return.
+	if len(ingress.Status.LoadBalancer.Ingress) == 0 ||
+		len(ingress.Status.LoadBalancer.Ingress[0].Hostname) == 0 {
+		logger.Debug("%s is ALB Ingress, but probably not provisioned, skip", ingress.Name)
+		return nil
+	}
+	// Expected e.g. bf647c9e-default-appingres-350b-1622159649.eu-central-1.elb.amazonaws.com where AWS ALB name is
+	// bf647c9e-default-appingres-350b (cannot be longer than 32 characters).
+	hostNameParts := strings.Split(ingress.Status.LoadBalancer.Ingress[0].Hostname, ".")
+	if len(hostNameParts) == 0 {
```

Comment:
I think `strings.Split` only returns an empty list if both the string and the separator are empty, this will never be true.

Reply:
Yes, you're right good catch, I want to ensure that 
```go
if len(hostNameParts[0]) == 0 
```